base/array.jl

```diff
@@ -892,6 +892,11 @@ end
 # set-like operators for vectors
 # These are moderately efficient, preserve order, and remove dupes.
 
+for T in [Bool,Int8,UInt8]
+    MAXSIZE = Int(typemax(T))-Int(typemin(T))+1
+    @eval unique(x::AbstractArray{$T}) = unique(x,$MAXSIZE)
```

Comment:
You don't need to use `eval` nor a loop AFAICT. Also `Int` does not appear to be required. Just write:
```julia
unique{T<:Union{Bool,Int8,UInt8}}(x::AbstractArray{T}) = unique(x, typemax(T)-typemin(T)+1)
```

(Same below.)